visualizer/main.py

```diff
@@ -0,0 +1,399 @@
+from google.auth.transport.requests import Request
+
+# If modifying these scopes, delete the file token.pickle.
+SCOPES = ['https://www.googleapis.com/auth/drive']
+
+class DataPoint:
+    pass
+
+class Parser:
+
+    def __init__(self, file_path, begin, end):
+        simulations = []
+        config_data = []
+        with open(file_path) as f:
+            for line in f:
+                # found start of section so start iterating from next line
+                if line.startswith(begin):
+                    simulation = []
+
+                    for line in f:
+                        # found end so end function
+                        if line.startswith(end):
+                            simulations.append(simulation)
+                            break
+
+                        elif line.startswith('*'):
+                            config_data.append(line)
+
+                        dp = self.string_to_datapoint(line)
+                        # yield every line in the section
+                        simulation.append(dp)
+
+        self.config = config_data
+        self.sims = simulations
+
+    def string_to_datapoint(self, line) -> DataPoint:
+
+        tokens = line.split(',')
+        type = tokens[0]
+        dp_params = tokens[1:]
+
+        if type == 'a':
+            return ArbiterDataPoint(arbiter_id=dp_params[0], consumer_id=dp_params[1], cycle_number=dp_params[2])
+        elif type == 'c':
+            return ConsumerDataPoint(consumer_id=dp_params[0], operation=dp_params[1], cycle_number=dp_params[2])
+
+class SheetRenderer:
+
+    def __init__(self, service, sims, config):
+        self.l_c_dict = {
+            "dark_yellow_1": (0.94, 0.76, 0.2),
+            "dark_green_1": (0.42, 0.66, 0.31),
+            "dark_cornflower_blue 1": (0.24, 0.5, 0.85),
+            "dark_red_1": (0.8, 0, 0),
+            'dark_purple_1': (0.4, 0.31, 0.65),
+            'dark_magenta_1': (0.65, 0.3, 0.47),
+            'dark_cyan_1': (0.27, 0.51, 0.56),
+            'dark_orange_1': (0.9, 0.57, 0.22),
+        }
+
+        self.d_c_dict = {
+            "dark_yellow_3": (0.49, 0.37, 0),
+            "dark_green_3": (0.15, 0.31, 0.07),
+            "dark_cornflower_blue 3": (0.11, 0.27, 0.53),
+            'dark_red_3': (0.4, 0, 0),
+            'dark_purple_3': (0.13, 0.01, 0.3),
+            'dark_magenta_3': (0.3, 0.07, 0.19),
+            'dark_cyan_3': (0.05, 0.21, 0.24),
+            'dark_orange_3': (0.5, 0.25, 0.02)
+        }
+
+        self.extra_light_colors = {
+            "light_yellow_1": (255, 217, 102),
+            "light_green_1": (147, 196, 125),
+            "light_yellow_3": (255, 242, 204),
+            "light_green_3": (217, 234, 211),
+        }
+
+        self.requests = []
+        self.service = service
+        self.sims = sims
+        self.config = config
+        self.offset_row = 0
+        self.start_of_con_in_sim = 0
+        self.start_of_arb_in_sim = 0
+
+    # find and register all unique arbiters
```

Comment:
Put the function level comments inside the function that they pertain too. 
You can also use docstrings:
"""
My description of what the function does.
"""